packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherPopover.tsx

```diff
@@ -60,12 +60,24 @@ export const OrganizationSwitcherPopover = React.forwardRef<HTMLDivElement, Orga
     }
 
     const handleOrganizationClicked = (organization: OrganizationResource) => {
-      return card.runAsync(() => setActive({ organization, beforeEmit: navigateAfterSwitchOrganization })).then(close);
+      return card
+        .runAsync(() =>
+          setActive({
+            organization,
+            beforeEmit: () =>
+              navigateAfterSwitchOrganization({
+                organization: organization,
```

Comment:
🙃 
**Nitpicky**:
```suggestion
                organization,
```
Also adding an explicit `return` might help with the formatting here.